src/System.Private.Reflection.Core/src/System/Reflection/Runtime/General/EcmaFormat/ReflectionTypeProvider.cs

```diff
@@ -0,0 +1,288 @@
+            RuntimeTypeInfo result = ((Handle)handle).TryResolve(reader, default(TypeContext), ref exception);
+            if (result != null)
+                return result;
+            else
+            {
+                ExceptionResult = exception;
+                return null;
+            }
+        }
+
+        // ISignatureTypeProvider
+        RuntimeTypeInfo ISignatureTypeProvider<RuntimeTypeInfo, TypeContext>.GetTypeFromSpecification(MetadataReader reader, TypeContext typeContext, TypeSpecificationHandle handle, byte rawTypeKind)
+        {
+            // raw Type Kind is either 0, ELEMENT_TYPE_CLASS, or ELEMENT_TYPE_VALUETYPE
+            Exception exception = null;
+            RuntimeTypeInfo result = ((Handle)handle).TryResolve(reader, typeContext, ref exception);
+            if (result != null)
+                return result;
+            else
+            {
+                ExceptionResult = exception;
+                return null;
+            }
+        }
+
+        RuntimeTypeInfo ISignatureTypeProvider<RuntimeTypeInfo, TypeContext>.GetFunctionPointerType(MethodSignature<RuntimeTypeInfo> signature)
+        {
+            return s_intPtrType;
+        }
+
+        RuntimeTypeInfo ISignatureTypeProvider<RuntimeTypeInfo, TypeContext>.GetGenericTypeParameter(TypeContext typeContext, int parameter)
+        {
+            if ((typeContext.GenericTypeArguments == null) ||
+                (typeContext.GenericTypeArguments.Length < parameter) ||
+                (parameter < 0))
+                ExceptionResult = new BadImageFormatException();
+
+            return typeContext.GenericTypeArguments[parameter];
+        }
+
+        RuntimeTypeInfo ISignatureTypeProvider<RuntimeTypeInfo, TypeContext>.GetGenericMethodParameter(TypeContext typeContext, int parameter)
+        {
+            if ((typeContext.GenericMethodArguments == null) ||
+                (typeContext.GenericMethodArguments.Length < parameter) ||
+                (parameter < 0))
+                ExceptionResult = new BadImageFormatException();
+
+            return typeContext.GenericMethodArguments[parameter];
+        }
+        
+        RuntimeTypeInfo ISignatureTypeProvider<RuntimeTypeInfo, TypeContext>.GetModifiedType(RuntimeTypeInfo modifier, RuntimeTypeInfo unmodifiedType, bool isRequired)
+        {
+            // Reflection doesn't really model custom modifiers...
+            return unmodifiedType;
+        }
+
+        RuntimeTypeInfo ISignatureTypeProvider<RuntimeTypeInfo, TypeContext>.GetPinnedType(RuntimeTypeInfo elementType)
+        {
+            // Reflection doesn't model pinned types
+            return elementType;
+        }
+
+        // IConstructedTypeProvider
+        RuntimeTypeInfo IConstructedTypeProvider<RuntimeTypeInfo>.GetGenericInstantiation(RuntimeTypeInfo genericType, ImmutableArray<RuntimeTypeInfo> typeArguments)
+        {
+            Type[] typeArgumentsAsType = new Type[typeArguments.Length];
```

Comment:
Why not just call directly into the TypeUnifier (genericType.GetConstructedGenericType(...)) like you did for the other parameterized types?